prime-router/src/main/kotlin/cli/tests/Hl7Tests.kt

```diff
@@ -55,6 +57,54 @@ class Hl7Ingest : CoolTest() {
             passed = passed and examineLineageResults(reportId, receivers, itemCount)
         }
 
+        return passed
+    }
+}
+
+class BadHl7 : CoolTest() {
+    override val name = "badhl7"
+    override val description = "Submit badly formatted hl7 files - should get errors"
+    override val status = TestStatus.SMOKE
+    override fun run(environment: ReportStreamEnv, options: CoolTestOptions): Boolean {
+        val listBadCharacters = listOf(
+            "ÅÍÎÏ˝ÓÔ\uF8FFÒÚÆ☃",
+            "OBR|1|A241Z^MESA_ORDPLC||P2^Procedure 2^ERL_MESA|||||||||xxx||Radiology^^^^R|7101^ESTRADA^JAIME^P^^DR|||||||||||1^once^^20000701^^R|||WALK|Modality Test 241||||||||||A||\n",
+            "PID|||||Richards^Mary||19340428|F|||||||||||||||||||\nPID|||||||19700510105000|M|||||||||||||||||||\n", /* multiple patient information*/
+            "~!@#\\\$^&*()-_=+[]\\\\{}|;':,./<>?",
+            "❤️ \uD83D\uDC94 \uD83D\uDC8C \uD83D\uDC95 \uD83D\uDC9E \uD83D\uDC93",
+            "<a href=\"javascript\\x0A:javascript:alert(1)\" id=\"fuzzelement1\">test</a>",
+            "'; EXEC sp_MSForEachTable 'DROP TABLE ?'; --",
+            "1'000'000,00"
+        )
+        var passed = false
+        val sender = hl7Sender
+        listBadCharacters.forEachIndexed { i, badCharacters ->
+            ugly("Starting badcsv file Test $i: submitting with $badCharacters")
+            val reFile = FileUtilities.replaceText(
+                "./src/test/hl7_test_files/invalid.hl7",
+                "replaceMe",
+                "$badCharacters"
+            )
+
+            if (!reFile.exists()) {
+                error("Unable to find file ${reFile.absolutePath} to do badhl7 test")
+            }
+
+            val (responseCode, json) = HttpUtilities.postReportFile(
+                environment,
+                reFile,
+                sender,
+                options.key
+            )
+            TermUi.echo("Response to POST: $responseCode")
+
+            if (responseCode >= 400) {
```

Comment:
This is a bit brittle.  If the container/API is not running then you get a 500 and pass the test.  Make sure to check the contents of the json response to make sure errors were indeed found in the data and that the data was rejected.

Reply:
done